.drone.jsonnet

```diff
@@ -24,12 +16,13 @@ local r_env_vars = {
   "BABYLON_EXE_PATH": std.join("/", [temp_volume_dir, "bbi"]),
 };
 
-# mpn-dev image tags; first element will be used to build release
+# images specified as repo:tag; first element will be used to build release
 # https://console.aws.amazon.com/ecr/repositories/mpn-dev/?region=us-east-1
-local mpn_snapshots = [
-  "cran-latest",
-  "2020-05-27",
-  "2020-03-24",
+# generally, only the second element should be modified
+local ci_images = [
+  "mpn-dev:latest",      # latest MPN snapshot
+  "mpn-dev:2020-03-24",  # oldest compatible snapshot
+  "cran-latest:latest",  # latest MPN snapshot for MRG packages + current CRAN
```

Comment:
Maybe these comments are just confusing but can you explain a little bit more about what the first and third snapshots are, and how often they will be refreshed?

Reply:
The idea is that the package will be tested against three images, described in the table below. The maintainer should only have to modify the second element, because only that one is expected to vary by package.

image | purpose | can vary by package? | changes when?
------|---------|------------------------|--------------
latest MPN snapshot | all MRG packages should work with the latest snapshot | no | when new snapshot is released
oldest compatible MPN snapshot | define the range of snapshot compatibility (through the current snapshot) | yes | images will be rebuilt periodically, but R package versions will not change
"CRAN latest" | MRG packages from the latest snapshot, plus non-MRG packages from the current source, e.g., current CRAN release of `dplyr`; this is the "canary in the coal mine" that should alert us to breaking changes in non-MRG packages so that we can update the MRG package being tested | no | triggered by change in upstream source, e.g., new version released on CRAN, or new MPN snapshot

Reply:
For all of these, are we doing the "dev" subset that is only the packages we know are used dependencies for MRG packages, or are we doing the full snapshot? My impression was that we were switching to the former so that build would be faster. Is that right? If not, are we still planning to do that at some point?

> "CRAN latest" ... triggered by change in upstream source

I'm not sure I follow. Are we actually gonna be monitoring CRAN and rebuilding this whenever one of our dependencies updates? If so... cool. If not, I just want to make sure this isn't manual (like someone checking CRAN and manually triggering an update). Like, it would be cool for this happen at least once a week or so. Unless it's automatic in which case I love it.

Reply:
> For all of these, are we doing the "dev" subset that is only the packages we know are used dependencies for MRG packages, or are we doing the full snapshot? My impression was that we were switching to the former so that build would be faster. Is that right? If not, are we still planning to do that at some point?

Yes, these images include only a subset of external (CRAN, Bioconductor) packages, including the dependencies of MRG packages.

> > "CRAN latest" ... triggered by change in upstream source
> 
> I'm not sure I follow. Are we actually gonna be monitoring CRAN and rebuilding this whenever one of our dependencies updates? If so... cool. If not, I just want to make sure this isn't manual (like someone checking CRAN and manually triggering an update). Like, it would be cool for this happen at least once a week or so. Unless it's automatic in which case I love it.

Yes, the plan is to have a job that watches upstream sources and triggers a rebuild when a dependency changes. We're aiming to automate as much of this as possible, and in particular, the plan is definitely _not_ to have this check be manual.

Reply:
Sounds good. Thanks for clarifying.